canonical/grafana-agent-k8s-operator
# TLS

## Issue
Fixes #217


## Solution
Use cert_manager to enable TLS.


## Context
Note that this does not enable TLS for the machine charm. This is because we can't change the cli args of a snap service. This is not a problem because the machine charm does not expose any endpoints outside the machine.


## Testing Instructions
1. Deploy Charm
2. Relate to a logging client (such as zinc-k8s)
3. relate to self-signed-certs
4. check that the endpoints are using tls


## Release Notes
Grafana Agent Operator now supports TLS for it's loki_push_api endpoints.
